### ISSUE TITLE
autoSlidePause is un-paused when the slider is moused over

### ISSUE THREAD

**Issue by irandamay** [iosscripts/iosslider#239]
I am programmatically stopping my slider that is set up with "autoSlide: true" with the new autoSlidePause method.  However, when a user mouses over the slider and then mouses back out, the autoSliding is resumed.  

It seems like if you explicitly stop the autoSliding, it should stay off until you explicitly turn it back on.

**Comment by marcwhitbread** [comment]
You're right! Sounds like a bug to me. Let me take a look at the source and get a fix out.

**Comment by Polyrhythm** [comment]
This feature doesn't work for me at all. If I set a slider to `autoSlide: true` and then call `$(element).iosSlider('autoSlidePause')` bound to an event, I get no effect. I've confirmed that the event fires because I've put a debug right before the pause would get executed.

Furthermore, I called `autoSlideStart` (accidentally) on a slider that has `autoSlide: false` and it actually enabled autoSlide.

**Comment by marcwhitbread** [comment]
@Polyrhythm sorry to hear that. The autoSlidePause function is still working in the benchmark when clicked in my benchmarks. You can find in the _demos/_benchmarks/_general/index.html file under Autoslide (play/pause functions)

Also, to make sure you are implementing it correctly. Make sure that the "element" object that you are calling autoSlidePause with is the same as the element that was initialized with iosSlider. eg:

``` js
$('.iosSlider').iosSlider();
$('.iosSlider').iosSlider('autoSlidePause');
```

> Furthermore, I called autoSlideStart (accidentally) on a slider that has autoSlide: false and it actually enabled autoSlide.

This is intended behavior. Does it not seem reasonable to have a autoslider start enabled manually and not on page load? If you think otherwise, I would like to get your opinion and I might change it.

**Comment by Polyrhythm** [comment]
I'll test this again tomorrow more extensively and report back.

As for autoSlideStart enabling autoSlide, it probably works fine as you describe. I interpreted the autoSlide initialization variable as a global state that had ultimate say in whether a given slider had the ability to be an auto-slider. Docs covering the feature should clear any misunderstandings like mine up.

**Comment by Polyrhythm** [comment]
So i've narrowed down the problem. The pause feature works fine if `autoSlide: false` is set initially. That way, `autoSlidePlay` will enable autoslide and then it can be paused. However, if you set `autoSlide: true` then it doesn't appear to pause successfully.

**Comment by Polyrhythm** [comment]
``` javascript
$player.brightcoveVideo('onMediaEvent', 'PLAY', function(event) {
        console.log('play');
        $('.iosSlider_hero').iosSlider('autoSlidePause');
      });

      $player.brightcoveVideo('onMediaEvent', 'STOP', function(event) {
        console.log('pause');
        $('.iosSlider_hero').iosSlider('autoSlidePlay');
      });
    };

    //for main hero carousel
    $('.iosSlider_hero').iosSlider({
      snapToChildren: true,
      desktopClickDrag: true,
      keyboardControls: true,
      autoSlide: true,
      autoSlideTimer: 5000,
      infiniteSlider: true,
      onSlideChange: changeThumb
    });
```

**Comment by irandamay** [comment]
As another data point, I am using it fine with "autoSlide: true" set initially.  My pausing is bound to click events.  It is only the mousing in/out that is causing a problem for me.  The following works as expected as long as I am careful to not put my mouse over the slider. 

```
$('#ios-slider').iosSlider({
  snapToChildren: true,
  desktopClickDrag: true,
  infiniteSlider: true,
  autoSlide: true,
  navNextSelector: $('#next-slide'),
  navPrevSelector: $('#prev-slide')
});

$('.slide-link').click(function() {
  // <SNIP>
  var $slider = $('#ios-slider');
  $slider.iosSlider('goToSlide', slide);
  $slider.iosSlider('autoSlidePause');
  // <SNIP>
});
```

**Comment by irandamay** [comment]
Oops, I clicked the wrong button to comment!

**Comment by Polyrhythm** [comment]
My user has to have a mouse over the slide to pause it so we may be experiencing the same issue then. Again, though, works as expected (even with the mouse over the slide) if `autoSlide: false`.

**Comment by marcwhitbread** [comment]
Sorry guys! I have reproduced the issue you both are experiencing on my end so it's just a matter of making a code change now. I'm working on it. :)

**Comment by marcwhitbread** [comment]
I believe I have fixed both your issues in v1.2.9. Download the latest zip and let me know if you are still experiencing issues.

**Comment by Polyrhythm** [comment]
Works for me. :+1:

**Comment by helperdesk** [comment]
This issue has returned for me. I'm using v1.3.19. Exactly the same issue as originally described in this ticket.

I have autoSlide set to true, and I have a button setup for autoSlideToggleSelector. The pause/play feature of the toggle button works great, but as soon as I mouse over the slider, it resumes sliding even though I explicitly told it to pause.

**Comment by marcwhitbread** [comment]
@definika The issue is actually unique to autoSlideToggleSelector. This bug is referring to the autoSlidePause/Play public methods. Anyway, I have applied a fix to your issue in v1.3.20.
